src/app/state/transaction/saga.ts

```diff
@@ -181,6 +221,44 @@ export function* doTransaction(action: PayloadAction<TransactionPayload>) {
   }
 }
 
+function* getAllowanceDifference(amount: string, runtimeAddress: string) {
+  const allowances = yield* select(selectAccountAllowances)
+  const allowance = allowances.find((item: Allowance) => item.address === runtimeAddress)?.amount || 0
+  return new BigNumber(amount).minus(allowance)
+}
+
+export function* submitParaTimeTransaction(runtime: Runtime, amount: string, recipient: string) {
+  const accountAddress = yield* select(selectAccountAddress)
+  const nic = yield* call(getOasisNic)
+  const chainContext = yield* select(selectChainContext)
+  const signer = yield* getSigner()
+  const allowanceDifference = yield* call(getAllowanceDifference, amount, runtime.address)
+
+  if (allowanceDifference.gte(0)) {
```

Comment:
Prefer using base unit bigints where possible, and BigNumbers mainly for converting from inputs and API